exchanges/ftx/ftx_wrapper.go

```diff
@@ -1682,3 +1691,265 @@ func (f *FTX) GetFuturesPositions(ctx context.Context, a asset.Item, cp currency
+			InitialMarginRequirement:     decimal.NewFromFloat(positions[i].InitialMarginRequirement),
+			EstimatedLiquidationPrice:    decimal.NewFromFloat(positions[i].EstimatedLiquidationPrice),
+			CollateralUsed:               decimal.NewFromFloat(positions[i].CollateralUsed),
+			MarkPrice:                    decimal.NewFromFloat(positions[i].EntryPrice),
+			CurrentSize:                  decimal.NewFromFloat(positions[i].Size),
+			BreakEvenPrice:               decimal.NewFromFloat(positions[i].RecentBreakEvenPrice),
+			AverageOpenPrice:             decimal.NewFromFloat(positions[i].RecentAverageOpenPrice),
+			RecentPNL:                    decimal.NewFromFloat(positions[i].RecentPNL),
+			MarginFraction:               decimal.NewFromFloat(acc.MarginFraction * 100),
+			FreeCollateral:               decimal.NewFromFloat(acc.FreeCollateral),
+			TotalCollateral:              decimal.NewFromFloat(acc.Collateral),
+		}, nil
+	}
+	return nil, fmt.Errorf("unable to calculate position summary %w for %v %v", order.ErrPositionNotFound, request.Asset, request.Pair)
+}
+
+// GetOpenPositions returns all open positions
+func (f *FTX) GetOpenPositions(ctx context.Context, item asset.Item, startDate time.Time) ([]order.OpenPositionDetails, error) {
+	if !item.IsFutures() {
+		return nil, fmt.Errorf("%w '%s'", order.ErrNotFuturesAsset, item)
+	}
+	positions, err := f.GetPositions(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+	pairs := make([]currency.Pair, len(positions))
+	for x := range positions {
+		pairs[x] = positions[x].Future
+	}
+	r := &order.GetOrdersRequest{
+		StartTime: startDate,
+		EndTime:   time.Now(),
+		AssetType: item,
+		Pairs:     pairs,
+	}
+	orders, err := f.GetOrderHistory(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(orders, func(i, j int) bool {
+		return orders[i].Date.Before(orders[j].Date)
+	})
+	response := make([]order.OpenPositionDetails, len(positions))
+	for x := range positions {
+		openPositionDetails := order.OpenPositionDetails{
+			Exchange: f.Name,
+			Asset:    item,
+			Pair:     positions[x].Future,
+		}
+		for y := range orders {
+			if !positions[x].Future.Equal(orders[y].Pair) {
+				continue
+			}
+			if len(openPositionDetails.Orders) == 0 && orders[y].Amount != positions[x].OpenSize && orders[y].Price != positions[x].RecentAverageOpenPrice {
+				// this is not the opening order for tracking open positions
+				continue
+			}
+			if orders[y].Date.Before(startDate) {
+				continue
+			}
+			if orders[y].Price == 0 {
+				// sometimes FTX sends a null price
+				orders[y].Price = orders[y].AverageExecutedPrice
+			}
+			openPositionDetails.Orders = append(openPositionDetails.Orders, orders[y])
+		}
+		if len(openPositionDetails.Orders) == 0 {
+			return nil, fmt.Errorf("%w open order found for %v but no orders after %v-%v", order.ErrPositionNotFound, positions[x].Future, startDate, r.EndTime)
```

Comment:
I get closed positions come through and it seems like its not filtering correctly which then stops the update of managed positions internally for things that are open. Example below:

```
{Future:OKB-PERP Size:0 Side:buy NetSize:0 LongOrderSize:0 ShortOrderSize:0 Cost:0 EntryPrice:0 UnrealizedPNL:0 RealizedPNL:28.77243284 InitialMarginRequirement:0.5 MaintenanceMarginRequirement:0.03 OpenSize:0 CollateralUsed:0 EstimatedLiquidationPrice:0 RecentAverageOpenPrice:0 RecentPNL:0 RecentBreakEvenPrice:0 CumulativeBuySize:0 CumulativeSellSize:0}
```

Maybe above we can filter positions that have OpenSize == 0 so we don't fetch order history. 

Also in this error context the position was found but an open order was not found or matched to position.

Reply:
Sweet catch! When testing I only had open positions, so this will fix up that discrepancy